.github/workflows/make_release.yaml

```diff
@@ -0,0 +1,44 @@
+name: create a release with a bumped version number
+
+on:
+  workflow_dispatch:
+    inputs:
+      NumberIndex:
+        description: 'Index of version number that should be bumped (for version x.y.z we have 0:x,1:y,2:z, so the default 2 is a tiny release).'
+        required: true
+        default: '2'
+      releaseBody:
+        description: 'If you want the release to have a description'
+        required: false
+
+jobs:
+  installSetup:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Bump version
+      run: |
+        i=${{github.event.inputs.versionNumber}}
+        [[ "$i" == '0' ]] || [[ "$i" == '1' ]] || [[ "$i" == '2' ]] || (echo "index must be 0, 1, or 2"; exit 1)
```

Comment:
Are we in bash? The code below would suggest that.

In that case you could use `(( i >= 0 && i <= 2 ))`.

Reply:
This would only work if a number is given not for strings, right?

Reply:
bash does not distinguish between a number and a string that looks like a number:

```bash
$ foo=1
$ echo $((foo + 1))
2
$ foo="1"
$ echo $((foo + 1))
2
```

Reply:
Ah, I thought if github provided a version _number_ it could be relied to be one.

In that case we should perform the check for "number-ness" via:

```bash
$ check() { [[ "$1" =~ ^[0-9]*$ ]] && echo yay || echo nay; }
$ check 23
yay
$ check foobar
nay
$ check 3
yay
$ check lalala
nay
```

Reply:
The variable is an input by the user, see https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/ and the image:
![image](https://user-images.githubusercontent.com/11610860/122211042-971bb300-cea6-11eb-933f-ad6aa919470c.png)

I adapted your second idea to first check whether any non number entry is there, and then check according to your first test, see 875fd11